### ISSUE TITLE
Add option to ignore destructuring vars

### ISSUE THREAD

**Issue by danwang** [eslint/eslint#5908]
ES6 destructuring can be used to omit properties from an object. This is somewhat common in React:

```
const {
  customThing1,
  customThing2,
  ...childProps,
} = this.props;

// Use childProps but not customThing{1,2}
```

It'd be nice to have an option to mark variables used when the variable is part of a destructuring assignment with a spread operator. All variables except the spread should be marked as used.

**What version of ESLint are you using?**
v2.7.0

**What rule do you want to change?**
Two options: 
1. Change `no-unused-vars` by adding a boolean option `ignoreDestructuringVars`
2. Add a new rule (e.g., `es6-ignore-unused-destructuring-vars`)

**What code should be flagged as incorrect with this change?**
The following code are not considered warnings:
```js
/*eslint no-unused-vars: ["error", {"ignoreDestructuringVars": true}] */

// rest is used, so a and b do not need to be used directly
function markUsed(x) {
  return x;
}
const {a, b, ...rest} = {};
markUsed(rest);
```

The following code are considered warnings:

```js
/*eslint no-unused-vars: ["error", {"ignoreDestructuringVars": true}] */

// a and rest are not used, so they should be omitted
function markUsed(x) {
  return x;
}
const {a, b, ...rest} = {};
markUsed(b);
```

**What happens when the rule is applied to this code now?**
```js
/*eslint no-unused-vars: "error" */

// existing behavior; a,b are unused
function markUsed(x) {
  return x;
}
const {a, b, ...rest} = {};
markUsed(rest);
```

**Comment by alberto** [comment]
Duplicate of https://github.com/eslint/eslint/issues/4880
